script.js

```diff
@@ -37,68 +51,73 @@ const createRows =(data)=>{
-              localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(data)); // Update local storage
-    });
-    
-    // Add event to edit button for this row
+     tablebody.appendChild(row);
+      // Add event to edit button for this row
     const editButton = row.querySelector('.edit-btn');
     editButton.addEventListener('click', () => {
-        // Prompt user for new data
-        const newName = prompt("Enter new name:", user.name);
-        const newEmail = prompt("Enter new email:", user.email);
-        const newRole = prompt("Enter new role:", user.role);
-        
-        // Update the user object
-        if (newName) user.name = newName;
-        if (newEmail) user.email = newEmail;
-        if (newRole) user.role = newRole;
-
-        // Update the row
-        row.children[2].textContent = user.name; // Update name cell
-        row.children[3].textContent = user.email; // Update email cell
-        row.children[4].textContent = user.role; // Update role cell
-
-        localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(data)); // Update local storage
+    editrows(user, row);
     });
-});
+       // Add event to delete button for this row
+       const deleteButton = row.querySelector('.del-btn');
+       deleteButton.addEventListener('click', () => {
+          deleteUser(user, data, row) 
+      });
+  });
 };
-createRows(data);
+   createRows(data);//call createrow()function to create rows in table//
+
+  // create a function to edit rows 
+   function editrows(user, row){
+    // Prompt user for new data
+    const newName = prompt("Enter new name:", user.name);
+    const newEmail = prompt("Enter new email:", user.email);
+    const newRole = prompt("Enter new role:", user.role);
+        
+    // Update the user object
+    if (newName) user.name = newName;
+    if (newEmail) user.email = newEmail;
+    if (newRole) user.role = newRole;
+
+     // Update the row
+    row.children[2].textContent = user.name; // Update name cell
+    row.children[3].textContent = user.email; // Update email cell
+    row.children[4].textContent = user.role; // Update role cell
+    localStorage.setItem(userData, JSON.stringify(data)); // Update local storage
+    }
 
-  //adding event to search button
-  const searchbutton =document.querySelector(".searchbtn");
-  const searchText= document.querySelector(".search");
+   
 
-  searchbutton.addEventListener("click",()=>{
-      const searchword= searchText.value.toLowerCase(); //.value is used retrieves the current text entered in the input field.
-      const filteredData= data.filter((user)=>{
-        const field =[user.id,user.name,user.email,user.role];
-        return field.some(field=>field.toLowerCase().includes(searchword));
-      });
-      createRows(filteredData);
+  //create a function to delete rows
+  function deleteUser(user, data, row){
+    row.remove(); // Remove the row from the table
+    data=data.filter(u => u.id !== user.id); // Update the data
```

Comment:
get data from local storage using get item dont use data variable.

`function deleteUser(user, data, row){
    row.remove(); // Remove the row from the table
    let storedData = localStorage.getItem(userData);
    storedData=storedData.filter(u => u.id !== user.id); // Update the data
    localStorage.setItem(userData, JSON.stringify(data)); // Update local storage
    createRows(storedData);
  }`